apps/dotcom/client/src/fairy/actions/LabelActionUtil.ts

```diff
@@ -17,8 +31,17 @@ export class LabelActionUtil extends AgentActionUtil<LabelAction> {
 		if (!action.complete) return action
 
 		const shapeId = helpers.ensureShapeIdExists(action.shapeId)
-		if (!shapeId) return null
-
+		if (!shapeId || !this.agent?.editor) {
+			return null
+		}
+		const shape = this.agent.editor.getShape(`shape:${shapeId}` as TLShapeId)
```

Comment:
@mimecuvalo could you take a look at the changes here? https://github.com/tldraw/tldraw/pull/7102/files#r2526219226

Reply:
i dig it - thank you!